index.bs

```diff
@@ -1975,6 +1979,124 @@ The following extended attributes are applicable to operations:
         "void"
 </pre>
 
+<h5 id="idl-tojson-operation">toJSON</h5>
+
+By declaring a “toJSON” [=regular operation=],
+an [=interface=] specifies how to convert the objects that implement it to [=JSON types=].
+
+The “toJSON” [=regular operation=] is reserved for this usage.
+It must take zero arguments and return a [=JSON type=].
+
+The <dfn id="dfn-json-types" export lt="JSON type">JSON types</dfn> are:
+
+*   [=numeric types=],
+*   {{boolean}},
+*   [=string types=],
+*   [=nullable types=] whose [=nullable types/inner type=] is a [=JSON type=],
+*   [=annotated types=] whose [=annotated types/inner type=] is a [=JSON type=],
+*   [=union types=] where all of their [=member types=] are [=JSON types=],
+*   [=sequence types=] whose parameterized type is a [=JSON type=],
+*   [=dictionaries=] where all of their [=dictionary members|members=] are [=JSON types=],
+*   [=records=] where all of their [=map/values=] are [=JSON types=],
+*   [=object types=],
+*   [=interface types=] that have a “toJSON” operation declared on themselves or
+    one of their [=inherited interface|inherited=] or
+    [=consequential interface|consequential=] interfaces.
+
+How the “toJSON” [=regular operation=]
+is made available on an object in a language binding,
+and how exactly the [=JSON types=] are converted into a JSON string,
+is language binding specific.
+
+Note: In the ECMAScript language binding,
+this is done by exposing a “toJSON” method
```

Comment:
I assume we're not consistent here, but maybe methods should be `<code>toJSON</code>`, even if we keep the ugly convention of curly-quotes for operation identifiers? Not a big deal.